### PR TITLE
Added support for setting the Colorspace on the output image

### PR DESCRIPTION
**Pull request by lasselehtinen**
I have a use case where we would like to force the output to be in a certain color space. This PR will add support for it.

### CODE REVIEW

File: src/Pdf.php [hunk 1]
```diff
@@ -14,6 +16,8 @@ class Pdf
 
     protected $outputFormat = '';
 
+    protected $outputColorSpace = imagick::COLORSPACE_RGB;
```
Comment (freekmurze) [inline src/Pdf.php]: Shouldn't it be "Imagick" instead of "imagick"?

File: tests/PdfTest.php [hunk 2]
```diff
@@ -35,4 +38,15 @@ public function it_will_throw_an_exception_when_try_to_convert_to_an_invalid_fil
 
         (new Pdf($this->testFile))->setOutputFormat('bla');
     }
+
+    /** @test */
+    public function setting_colorspace_is_reflected_in_output_image()
+    {
+        $pdf = new Pdf($this->testFile);
+        $pdf->setPage(1)->setResolution('72')->setOutputColorSpace(imagick::COLORSPACE_CMYK)->saveImage($this->testImage);
```
Comment (freekmurze) [inline tests/PdfTest.php]: Use "Imagick" instead of "imagick"

File: src/Pdf.php [hunk 3]
```diff
@@ -67,6 +71,22 @@ public function setOutputFormat($outputFormat)
     }
 
     /**
+     * Set the output color space.
+     *
+     * @param string $outputColorSpace
+     *
+     * @return $this
+     *
+     * @throws \Spatie\PdfToImage\Exceptions\InvalidColorspace
```
Comment (freekmurze) [inline src/Pdf.php]: This exception isn't being thrown. Could you add this?

**Comment by freekmurze** [comment]
Hi, thanks for your contribution!

I added a few remarks. Could you please take a look at them?

**Comment by lasselehtinen** [comment]
My bad, I had the isValidOutputColorSpace method there at some point. I have trouble figuring out a problem with checking the if the colorspace is actually correct. For the colorspace, you are suppose to use the [Imagick constants](http://php.net/manual/en/imagick.constants.php) for Colorspace. If you use [defined](http://php.net/manual/en/function.defined.php) to check if it exists, it always returns false. I used [get_defined_constants](http://php.net/manual/en/function.get-defined-constants.php) to get list of defined constants, but I cannot find anything related to Imagick. And the extension is indeed loaded.

```
vagrant@homestead:~/Code/pdf-to-image$ php -i|grep imagick
/etc/php/7.0/cli/conf.d/20-imagick.ini,
imagick
imagick module => enabled
imagick module version => 3.4.0
imagick classes => Imagick, ImagickDraw, ImagickPixel, ImagickPixelIterator
imagick.locale_fix => 0 => 0
imagick.progress_monitor => 0 => 0
imagick.skip_version_check => 0 => 0
```

```
    /**
     * Determine if the given colorspace is a valid.
     *
     * @param $outputFormat
     *
     * @return bool
     */
    public function isValidOutputColorSpace($outputColorSpace)
    {        
        return defined($outputColorSpace);
    }
```

**Comment by lasselehtinen** [comment]
I haven't worked much on constants on PHP. I just realised that it is casted to the actual value (int) when passed to the method isValidOutputColorSpace. So it will not even see the constant name or anything, just the value. I will have to figure out some way to validate that the value. Any ideas?

**Comment by freekmurze** [comment]
If validating the color space proves too hard, I'm ok with dropping the exception alltogether. Then it's the user responsability to pass in a valid constant / value.

**Comment by lasselehtinen** [comment]
I've changed so that the colorspace is passed as a string. Then we can check if the constant actually exists. It also uses the Imagick class prefix to check, so you cannot pass some other constant that just happens to have the same value. Somehow this feels little bit hacky, but at least there is a some kind of check in place.

**Comment by freekmurze** [comment]
Merge conflicts, could you rebase to the current master?

**Comment by lasselehtinen** [comment]
Done, but seems like now unit tests fail on Travis. It works fine on my two different enviroments, so I assume it is some kind of issue with the Imagick extension on Travis. Or maybe overwriting the existing "test.jpg". I will try to dig in to this.

File: src/Pdf.php [hunk 4]
```diff
@@ -168,8 +204,14 @@ public function getImageData($pathToImage)
 
         $imagick->readImage(sprintf('%s[%s]', $this->pdfFile, $this->page - 1));
 
+        $imagick->setImageFormat($this->determineOutputFormat($pathToImage));
+
+        $imagick->transformImageColorspace($this->outputColorSpace);
+
         $imagick->setFormat($this->determineOutputFormat($pathToImage));
 
+        file_put_contents($pathToImage, $imagick);
```
Comment (jamesdb) [inline src/Pdf.php]: I have issues with this line, see this previous PR https://github.com/spatie/pdf-to-image/pull/4 for more details.

This makes assumptions about how people are storing files.

**Comment by freekmurze** [comment]
Closing due to inactivity

**Pull request closed without merging**